src/view/view.cpp

```diff
@@ -403,7 +403,13 @@ void wf::view_interface_t::set_fullscreen(bool full)
     if (!fullscreen && get_output() && needs_lowering)
         get_output()->workspace->add_view(self(), wf::LAYER_WORKSPACE);
 
-    this->emit_signal("fullscreen", nullptr);
+    view_fullscreen_signal data;
+    data.view = self();
+    data.state = full;
+    data.desired_size = get_output()->get_relative_geometry();
+
+    get_output()->emit_signal("view-fullscreen", &data);
+    this->emit_signal("fullscreen", reinterpret_cast<signal_data_t*> (full));
```

Comment:
Maybe we can just emit the same data as for the `view-fullscreen` signal and change any previous "subscribers"